Sources/Fluent/Model/Model.swift

```diff
@@ -258,8 +258,14 @@ extension Model where Database: QuerySupporting, ID: KeyStringDecodable {
         }
 
         let dbid = try Self.requireDefaultDatabase()
-        return container.requestCachedConnection(to: dbid).flatMap(to: Self.self) { conn in
-            self.find(id, on: conn).map(to: Self.self) { model in
+        let aConn : Future<Database.Connection>
+        if let subcontainer = container as? SubContainer {
+            aConn = subcontainer.requestCachedConnection(to: dbid)
+        } else {
+            aConn = container.requestPooledConnection(to: dbid)
```

Comment:
Need to either use `.withPooledConnection` or call `.releasePooledConnection` once the find has finished.

You can make use of a nested function here to prevent code-dupe. 

(`requestCachedConnection` is fine not calling any release because it's assumed that a subcontainer will call release when it deallocates)

Reply:
Cool, that makes sense. Take a look at my changes and let me know if that's what you're thinking about for the nested function 👍